src/Data/SafeJSON/Internal.hs

```diff
@@ -268,17 +274,17 @@ getVersion _ = Nothing
 -- "{\"~v\":0,\"~d\":\"test\"}"
 --
 -- @since 1.0.0
-setVersion' :: forall a. SafeJSON a => Version a -> Value -> Value
+setVersion' :: forall a. SafeJSON a => Version a -> SafeValue -> SafeValue
```

Comment:
I designed the `setVersion` functions to mainly be used to set the version when the value does not have one (taking a format from a third party who doesn't need to know about the versioning, for example)

So I'd say it would be more logical for the type to be `Value -> SafeValue`